### PR TITLE
fix a incompatible pointer type warning for gcc14

### PR DESCRIPTION
**Pull request by raveit65**
closes https://github.com/mate-desktop/atril/issues/603

### CODE REVIEW

**Review by lukefromdc — APPROVED** [review]
Same code as my second earlier test, gave same behavior as master and suppressed the warnings. I actually agree with gcc-14 treating incompatable pointers as an error by default, because my experience is code with that problem very often doesn't run right. This one with just a conflict over const seemed to work OK in master but that's the exception not the rule. Good to fix this instead of relying on compiler behavior.

Applying both this and https://github.com/mate-desktop/atril/pull/600 together stopped ALL the build warnings with default cflags and gcc-13 on Debian Unstable

File: backend/epub/epub-document.c [hunk 1]
```diff
@@ -632,7 +632,7 @@ check_mime_type(const gchar* uri,GError** error)
         const gchar* mimetypes[] = {"application/epub+zip", "application/x-booki+zip", NULL};
         guint i;
 
-        for (i = 0; i < g_strv_length (mimetypes); i++) {
+        for (i = 0; i < g_strv_length ((gchar**) mimetypes); i++) {
```
Comment (cwendling) [inline backend/epub/epub-document.c:635]: could also be fixed and optimized using
```suggestion
        for (i = 0; mimetypes[i]; i++) {
```

or
```suggestion
        for (i = 0; i < G_N_ELEMENTS (mimetypes); i++) {
```

But that's kind of beyond the scope of this fix.

**Review by cwendling — APPROVED** [review]
LGTM, if that's all there is to fixing the issue :+1:

File: backend/epub/epub-document.c [hunk 2]
```diff
@@ -643,7 +643,7 @@ check_mime_type(const gchar* uri,GError** error)
             mimeFromFile = ev_file_get_mime_type (uri, TRUE, &err);
             if (mimeFromFile)
             {
-                for (i = 0; i < g_strv_length (mimetypes); i++) {
+                for (i = 0; i < g_strv_length ((gchar**) mimetypes); i++) {
```
Comment (cwendling) [inline backend/epub/epub-document.c:646]: If you choose this solution, let's apply it here as well
```suggestion
                for (i = 0; mimetypes[i]; i++) {
```

**Comment by raveit65** [comment]
wtf, `commit suggest` from conversation did only add one line.

**Comment by cwendling** [comment]
@raveit65 I didn't add it to the second occurrence because I didn't think it was a necessary change, but merely a possible improvement; and you can only have one suggestion per individual hunk, so I couldn't do both at once.  It's no biggie, but maybe we want to harmonize this in a follow-up :)

**Comment by raveit65** [comment]
I pushed a an update to master :-)

**Comment by cwendling** [comment]
@raveit65 I see :)  Though, I'd rather use the same technique in both instance, rather than 2 different ones

**Comment by raveit65** [comment]
It was a kind of bad website automatism. After i added the first suggestion github marked/closed the conversation as resolved and i thought both suggestions were added.
Than i squashed and merged the PR without checking the commit again.
...good i learnt how a small github detail is working :)

**Pull request merged**